app/models/topic_link.rb

```diff
@@ -185,9 +186,10 @@ def self.ensure_entry_for(post, link, parsed)
 
       topic_id = route[:topic_id].to_i
       post_number = route[:post_number] || 1
+      topic_slug = route[:id]
 
       # Store the canonical URL
-      topic = Topic.find_by(id: topic_id)
+      topic = Topic.find_by("id = ? OR slug = ?", topic_id, topic_slug)
```

Comment:
oh the index is on `topic_slug is not null` so this can trigger a table scan... instead double query is probably correct here

```
topic = Topic.find_by(id: topic_id)
if !topica && topic_slug
    topic = Topic.find_by(slug: topic_slug)
end
```